src/Action/RetrieveAutocompleteItemsAction.php

```diff
@@ -100,10 +103,16 @@ public function __invoke(Request $request): JsonResponse
         }
 
         $searchText = $request->get('q', '');
+        if (!\is_string($searchText)) {
+            throw new BadRequestParamHttpException('q', 'string', $searchText);
+        }
 
         $targetAdmin = $fieldDescription->getAssociationAdmin();
 
         // check user permission
+        if (!\is_string($targetAdminAccessAction)) {
+            throw new \RuntimeException('The "target_admin_access_action" action must be a string.');
```

Comment:
```suggestion
            throw new \TypeError('The "target_admin_access_action" action must be a string.');
```
Also, I think "attribute" could be better than "action" in this sentence, given the way we are obtaining this value.

Reply:
I would say I don't like throwing a TypeError or a RuntimeException in an action since it results to a 500.

Wouldn't it better to throw an HttpException to be handled by Symfony ?
Does BadRequestHttpException is the best ? it would throw a 400.

Reply:
Yes, I agree than an HTTP exception would be far better than an unhandled exception.